pkg/combustion/kubernetes_test.go

```diff
@@ -0,0 +1,251 @@
+package combustion
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/suse-edge/edge-image-builder/pkg/fileio"
+	"github.com/suse-edge/edge-image-builder/pkg/image"
+)
+
+type mockKubernetesScriptInstaller struct {
+	installScript func(distribution, sourcePath, destPath string) error
+}
+
+func (m mockKubernetesScriptInstaller) InstallScript(distribution, sourcePath, destPath string) error {
+	if m.installScript != nil {
+		return m.installScript(distribution, sourcePath, destPath)
+	}
+
+	panic("not implemented")
```

Comment:
I'm personally not a fan of any sort of panic/exit in the middle of unit tests; I think they should always run to completion. It's similar in my head to using `assert` instead of `require` where it makes sense. But it's not a big enough problem to warrant blocking the merge.

Reply:
It is in fact a sanity check which prevents a misuse of the mock itself. We often rely on the returned error as a value to determine whether or not the mock operation is successful so distinguishing intentional errors with a panic is usually the way to go in these cases.